location/templates/location.html

```diff
@@ -22,5 +23,41 @@
 
 {% block content %}
 <h1>{{object.address}}, {{object.city}}, {{object.state}}, {{object.zipcode}}</h1>
+    # After onclick "write review...",  should POST and display write review
+    # should automatically update the page
 {% leaflet_map "leaflet_map_main" %}
+    {% if object.review_set %}
+        <h2> Review: </h2>
+        <hr class="mt-0 mb-2 mt-2">
+        <div class="row h-65">
+            <div class="col">
+                <table class="table table-striped table-bordered table-hover table-sm results-table" width="100%">
+                    <thead>
+                        <tr>
+                            <th scope="col" class="th-sm">User</th>
+                            <th scope="col" class="th-sm">Content</th>
+                            <th scope="col" class="th-sm">Time</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {% for review in object.review_set.all %}
+                        <tr>
+                            <td scope="row">{{ review.user }}</td>
+                            <td>{{ review.content }}</td>
+                            <td>{{ review.time }}</td>
+                        </tr>
+                        {% endfor %}
+                    </tbody>
+                </table>
+            </div>
+        </div>
+    {% endif %}
+    <h2>Write some review</h2>
```

Comment:
currently this displays regardless of whether or not I'm logged in and if I am  "Tenant". I think at the very least we should check to see if the user is currently logged in and if they are a tenant or not.

Reply:
its nice that it redirects you to the login page if you try to submit a review, but I think a user that took time to write out a long, well thought out review would be mad if they clicked submit and had their entire review thrown away because they forgot to log in. 

 I think the better option would be to have a button "Write a Review" that would be visible only when a tenant is logged in, and then when they clicked it it would show (i.e., un-hide) the review submission form similar to how Muhammad hides/shows the 311 statistic data.  That way the review creation functionality is hidden from the wrong people, and for those that are tenants the review form doesn't unnecessarily clutter up the screen.

Reply:
Make sense!

Reply:
Completely agree with what Alex said. Also I think it would make for better viewing to have the "Write a review" on top and the reviews below that which would be shown/hidden based on button click.